elements/lisk-api-client/src/block.ts

```diff
@@ -12,40 +12,135 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import { Channel, RegisteredSchemas } from './types';
-import { decodeBlock, encodeBlock } from './codec';
+import { codec } from '@liskhq/lisk-codec';
+import { Block as BlockType, Channel, RegisteredSchemas } from './types';
+import { decodeBlock, encodeBlock, getTransactionAssetSchema } from './codec';
 
 export class Block {
 	private readonly _channel: Channel;
-	private readonly _schema: RegisteredSchemas;
+	private readonly _schemas: RegisteredSchemas;
 
 	public constructor(channel: Channel, registeredSchema: RegisteredSchemas) {
 		this._channel = channel;
-		this._schema = registeredSchema;
+		this._schemas = registeredSchema;
 	}
 
 	public async get(id: Buffer): Promise<Record<string, unknown>> {
 		const blockHex = await this._channel.invoke<string>('app:getBlockByID', {
 			id: id.toString('hex'),
 		});
 		const blockBytes = Buffer.from(blockHex, 'hex');
-		return decodeBlock(blockBytes, this._schema);
+		return decodeBlock(blockBytes, this._schemas);
 	}
 
 	public async getByHeight(height: number): Promise<Record<string, unknown>> {
 		const blockHex = await this._channel.invoke<string>('app:getBlockByHeight', { height });
 		const blockBytes = Buffer.from(blockHex, 'hex');
-		return decodeBlock(blockBytes, this._schema);
+		return decodeBlock(blockBytes, this._schemas);
 	}
 
 	public encode(input: {
 		header: Record<string, unknown>;
 		payload: Record<string, unknown>[];
 	}): Buffer {
-		return encodeBlock(input, this._schema);
+		return encodeBlock(input, this._schemas);
 	}
 
 	public decode(input: Buffer): Record<string, unknown> {
-		return decodeBlock(input, this._schema);
+		return decodeBlock(input, this._schemas);
+	}
+
+	public toJSON(
+		block: BlockType,
+	): {
+		header: Record<string, unknown>;
+		payload: Record<string, unknown>[];
+	} {
+		const json = {
+			header: {
+				asset: {},
+			},
+			payload: [],
+		};
+		const { asset, ...headerRoot } = block.header;
+
+		// We need to remove this 'cos schema does not have id
+		delete headerRoot.id;
+
+		// decode header
+		json.header = { ...codec.toJSON(this._schemas.blockHeader, headerRoot), asset: {} };
```

Comment:
if u reassign whole object, maybe it would be better to just create new variable here?
like
```
const header = { ...codec.toJSON(this._schemas.blockHeader, headerRoot), asset: {} };
```